### ISSUE TITLE
Expose commit (aka private bytes) metric for a function app through MDM

### ISSUE THREAD

**Issue by paulbatum** [Azure/Azure-Functions#726]
GB-sec usage for a consumption function app is calculated based on how long a function runs for and how much memory was consumed by the process during that time. Specifically, commit / private bytes. This memory usage is limited to 1.5GB per instance.

Ideally, the commit for the process would be captured as a metric and exposed through MDM. This would allow customers to gain an understanding of their memory usage and to create alerts. The latter capability is particularly useful when you consider that the runtime will start throwing OutOfMemoryExceptions once these limits are hit. These exceptions are hard to handle gracefully and some customers would prefer to be notified through these alerts so they can take action in their code/config to reduce the memory footprint of their function app.

By making this metric available through MDM, it would be browsable from the Monitor->Metrics UX in the azure portal.

**Comment by ColbyTresness** [comment]
@nimakms for monitoring
@paulbatum is this still relevant?

**Comment by paulbatum** [comment]
@ColbyTresness Yes its relevant. Not sure why you put this in the active questions milestone. It describes a known platform gap.

**Comment by ColbyTresness** [comment]
Will move to backlog.

**Comment by paulbatum** [comment]
Actually, I just realized that functions V2 is completely missing performance counters (I just filed https://github.com/Azure/azure-functions-host/issues/3852 to track). So I have marked this as blocked. We'd need to resolve that first.

**Comment by marcduiker** [comment]
Just curious: is this issue still blocked? Due to https://github.com/Azure/Azure-Functions/issues/1148 (Linux Metrics) instead of https://github.com/Azure/azure-functions-host/issues/3852?

**Comment by paulbatum** [comment]
Yes it should be resolved on Windows and is still blocked on Linux.
